lib/net/dhcp/ztp_record.rb

```diff
@@ -0,0 +1,19 @@
+module Net::DHCP
+  class ZTPRecord < Record
+    attr_accessor :firmware, :vendor
+
+    def initialize(opts = { })
+      super(opts)
+      self.firmware    = opts[:firmware]
+      self.vendor      = opts[:vendor]
+    end
+
+    def attrs
+      super.merge(
+        { 
+          :ztp_firmware   => firmware,
+          :ztp_vendor     => vendor
+        }).delete_if { |k, v| v.nil? }
```

Comment:
You should be able to use `compact` here.

Reply:
Good one, thank you!